spec.md

```diff
@@ -325,19 +330,28 @@ on the definition of OPTIONAL.
   event format defines the relationship in
   [section 3.1](./json-format.md#31-handling-of-data).
 
-  When this attribute is omitted, `data` simply follows the event
-  format's encoding rules. For the JSON event format, the `data` value
-  can therefore be a JSON object, array, or value.
-
-  For the binary mode of some of the CloudEvents transport bindings, where the
-  `data` content is immediately mapped into the payload of the transport frame,
-  this field is directly mapped to the respective transport or application
+  For the binary mode of some of the CloudEvents protocol bindings, where the
+  `data` content is immediately mapped into the payload of the protocol frame,
+  this field is directly mapped to the respective protocol or application
   protocol's content-type metadata property. Normative rules for the binary mode
-  and the content-type metadata mapping can be found in the respective transport
+  and the content-type metadata mapping can be found in the respective protocol
   mapping specifications.
 
+  This attribute MAY be omitted if the event format has an implied
```

Comment:
Use of the word "implied" here can be confusing. If I'm understanding things, this is really talking about the case where it's a "structured" message and therefore w/o a datacontenttype attribute we can assume that `data` just follows the format of the payload itself, right?  If so, would this wording be ok:

```
In structured mode, where the CloudEvents attributes are siblings to `data` in the message
payload, this attribute MAY be omitted when the format of `data` is the same as the format
of the message itself. For example, when the payload is of type JSON, `datacontenttype` can
be omitted when `data` is also of type JSON.
```

Reply:
Agreed it's confusing and again the use of terms "binary" and "structured" is causing us problems. Structured mode *messages* are not events an do not have a datacontentype attribute. They *contain* an event in some in some "application/cloudevents+" encoding (or multiple events now we have batch formats). 

Only when you decode an individual event  do you see a dataconenttype attribute - and at that point it is irrelevant whether the event arrived wrapped in a structured-mode message, as part of a batch, or directly (e.g. JSON events read from stdin.)

So this issue applies only to *directly-encoded* events. We use the term "binary" for HTTP and AMQP directly-encoded events, we don't for JSON since that would exacerbate the base64 confusion. "binary" and "structured" are really just the wrong terms. I'll take another shot avoiding these terms and "implied'